frame/staking/src/lib.rs

```diff
@@ -1270,6 +1453,100 @@ impl<T: Trait> Module<T> {
 
 	// MUTABLES (DANGEROUS)
 
+	fn do_payout_nominator(who: T::AccountId, era: EraIndex, validators: Vec<(T::AccountId, u32)>)
+		-> DispatchResult
+	{
+		let mut nominator_ledger = <Ledger<T>>::get(&who).ok_or_else(|| Error::<T>::NotController)?;
+		if nominator_ledger.next_reward > era {
+			return Err(Error::<T>::InvalidEraToReward.into());
+		}
+
+		nominator_ledger.next_reward = era + 1;
+		<Ledger<T>>::insert(&who, &nominator_ledger);
+
+		let mut reward = Perbill::zero();
+		let era_reward_points = <ErasRewardPoints<T>>::get(&era);
+
+		// Note: iterator take only `MAX_NOMINATIONS` to avoid querying more validator exposure
+		// than necessary. Anyway a nominator can't validate more than `MAX_NOMINATIONS`.
+		for (validator, nominator_index) in validators.into_iter().take(MAX_NOMINATIONS) {
```

Comment:
Why not just return an error if validators length is greater than `MAX_NOMINTATIONS`?

Might be more user friendly.

Reply:
yes indeed it doesn't cost anything, done